### PR TITLE
New Framework & Profile images

### PR DESCRIPTION
**Pull request by andrewxhill**

### CODE REVIEW

File: App/Config/ReduxPersist.js [hunk 1]
```diff
@@ -66,14 +66,14 @@ const REDUX_PERSIST = {
   reducerVersion: '1.0',
   storeConfig: {
     key: 'primary',
-    version: 4,
+    version: 5,
```
Comment (asutula) [inline App/Config/ReduxPersist.js]: I don't think you need to increment this if you're not providing any migration.

**Comment by andrewxhill** [comment]
@asutula I added 'contacts' to the persist whitelist, take a look and see if okay with that

File: App/Redux/ThreadsRedux.ts [hunk 2]
```diff
@@ -213,6 +213,7 @@ export function reducer (state: ThreadsState = initialState, action: ThreadsActi
 }
 
 export const ThreadsSelectors = {
+  threads: (state: any) => state.threads,
```
Comment (asutula) [inline App/Redux/ThreadsRedux.ts]: Dup?

File: App/Config/ReduxPersist.js (hunk 1 above)
Comment (andrewxhill) [reply App/Config/ReduxPersist.js]: undone

File: App/Redux/ThreadsRedux.ts (hunk 2 above)
Comment (andrewxhill) [reply App/Redux/ThreadsRedux.ts]: gone

File: App/Sagas/TextileSagas.ts [hunk 3]
```diff
@@ -97,9 +96,14 @@ export function * handleProfilePhotoSelected(action: ActionType<typeof UIActions
     }
     const blockId: string = yield call(TextileNode.addPhotoToThread, addResult.id, addResult.key, defaultThread.id)
     yield put(UploadingImagesActions.addImage(addResult.archive.path, addResult.id, 3))
+
+
     yield put(TextileNodeActions.getPhotoHashesRequest(defaultThread.id))
 
-    //TODO: Call some TextileNode method to add profile picture
+    // set it as our profile picture
+    yield call(TextileNode.setAvatarId, addResult.id)
```
Comment (asutula) [inline App/Sagas/TextileSagas.ts]: Nice.

File: App/Config/ReduxPersist.js [hunk 4]
```diff
@@ -73,7 +73,7 @@ const REDUX_PERSIST = {
     // blacklist: ['nav', 'ipfs', 'auth', 'ui'],
     // Optionally, just specify the keys you DO want stored to persistence.
     // An empty array means 'don't store any reducers' -> infinitered/ignite#409
-    whitelist: ['preferences', 'uploadingImages', 'cameraRoll'],
+    whitelist: ['preferences', 'uploadingImages', 'cameraRoll', 'contacts'],
```
Comment (asutula) [inline App/Config/ReduxPersist.js]: Yea looks good for now. Doesn't seem like a good long term plan to rely on this persistence if this is the only place we're keeping track of contacts (like this should somehow be stored in a decentralized way?)

**Pull request merged**